recipes/coost/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_libcurl": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libcurl": False,
+        "with_openssl": False,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        if self.options.with_libcurl:
+            self.requires("libcurl/8.2.1")
+        if self.options.with_libcurl or self.options.with_openssl:
+            self.requires("openssl/[>=1.1 <4]")
+        if self.settings.os == "Linux":
+            self.requires("libbacktrace/cci.20210118")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        if self.info.options.with_libcurl:
+            if not self.info.options.with_openssl:
+                raise ConanInvalidConfiguration(f"{self.ref} requires with_openssl=True when using with_libcurl=True")
+            if self.dependencies["libcurl"].options.with_ssl != "openssl":
+                raise ConanInvalidConfiguration(f"{self.ref} requires libcurl/*:with_ssl='openssl' to be enabled")
+            if not self.dependencies["libcurl"].options.with_zlib:
+                raise ConanInvalidConfiguration(f"{self.ref} requires libcurl/*:with_zlib=True to be enabled")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if is_msvc(self):
+            tc.variables["STATIC_VS_CRT"] = is_msvc_static_runtime(self)
+        tc.variables["WITH_LIBCURL"] = self.options.with_libcurl
+        tc.variables["WITH_OPENSSL"] = self.options.with_openssl
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "coost")
+        self.cpp_info.set_property("cmake_target_name", "coost::co")
+        # TODO: back to global scope in conan v2 once legacy generators removed
```

Comment:
Thanks a lot for the contribution! 
I think that being this a new recipe that is not adding test_v1 to test the legacy generators, we could get rid of the component.

Reply:
@czoido 
Thank you for reviewing it.
I forgot to delete it.

Reply:
fixed.